src/test/surefire.go

```diff
@@ -10,31 +10,22 @@ import (
 
 // CopySurefireXmlFilesToDir copies all the XML test results files into the given directory.
 func CopySurefireXmlFilesToDir(state *core.BuildState, surefireDir string) {
-	outputDirs := make(map[string]struct{})
 	for _, label := range state.ExpandOriginalLabels() {
 		target := state.Graph.TargetOrDie(label)
 		if state.ShouldInclude(target) && target.IsTest && !target.NoTestOutput {
-			outputDir := target.OutDir()
-			if !core.PathExists(outputDir) {
-				// Unable to find tests
-				continue
-			}
-			outputDirs[outputDir] = struct{}{}
-		}
-	}
-
-	for outputDir := range outputDirs {
-		fs.Walk(outputDir, func(path string, isDir bool) error {
-			if !isDir {
-				bytes, _ := ioutil.ReadFile(path)
-				if looksLikeJUnitXMLTestResults(bytes) {
-					surefireResult := filepath.Join(surefireDir, filepath.Base(path))
-					if err := fs.CopyOrLinkFile(path, surefireResult, 0644, true, true); err != nil {
-						log.Errorf("Error linking %s to %s - %s", surefireResult, path, err)
+			if path := target.TestResultsFile(); fs.PathExists(path) {
+				fs.Walk(path, func(path string, isDir bool) error {
```

Comment:
Do we still need to walk, or is testResultsFile really a directory?

Reply:
It really is a directory for both Python and Java tests; they write one file per class or smthn.

Found that out the hard way, the first version didn't walk and I didn't get many results files :(

Reply:
Alright, LGTM